src/coreclr/dlls/mscoree/coreclr/CMakeLists.txt

```diff
@@ -1,3 +1,10 @@
+
+# we need to define this library before adding .def fiels to sources.
```

Comment:
I wonder why it wasn't needed before.

Reply:
This is ignored by a static lib, but becomes an error if it is `coreclr_static` is an object lib. 
Same about duplicate inclusion of `gcdecode.cpp`, which causes a warning with static lib and an error with object lib. 

Basically static lib can tolerate these and we do not _have_ to fix. 
I had to do the fixes for the object lib, but even though we are not using it in this change, I kept the fixes, since these things (like including .def files) do not look intentional.